test/unit/teacher_set_test.rb

```diff
@@ -182,13 +193,293 @@ class TeacherSetTest < ActiveSupport::TestCase
+      @mintest_mock1.expect(:delete_document_by_id, expected_resp, [nil])
+      resp = nil
+      resp = assert_raises(StandardError) do
+        ElasticSearch.stub :new, @mintest_mock1 do
+          resp = @teacher_set.delete_teacherset_record_from_es
+        end
+      end
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:code], resp.code)
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:msg], resp.message)
+    end
+  end
+
+
+  # test teacher set object method.
+  describe "test teacher set object" do
+    it 'test teacher set object' do
+      resp = @teacher_set.teacher_set_info
+      assert_equal(@teacher_set.title, resp[:title])
+      assert_equal(@teacher_set.bnumber, resp[:bnumber])
+      assert_equal(@teacher_set.subjects.count, resp[:subjects].count)
+    end
+  end
+
+
+  describe 'update set_type value in teacher_set table' do
+    it 'test update_set_type value method' do
+      resp = nil
+      set_type_val = @teacher_set3.set_type
+      @mintest_mock1.expect(:call, set_type_val, [set_type_val])
+      @mintest_mock2.expect(:call, true, [set_type: set_type_val])
+
+      @teacher_set3.stub :derive_set_type, @mintest_mock1 do
+        @teacher_set3.stub :update, @mintest_mock2 do
+          resp = @teacher_set3.update_set_type(set_type_val)
+        end
+      end
+      assert_equal(set_type_val, resp)
+      @mintest_mock1.verify
+      @mintest_mock2.verify
+    end
+
+    it 'test StandardError while updating the set_type' do
+      resp = nil
+      teacher_set = TeacherSet.new
+      set_type = "Topic Set"
+      @mintest_mock1.expect(:call, set_type)
+
+      resp = assert_raises(StandardError) do
+        teacher_set.stub :derive_set_type, @mintest_mock1 do
+          resp = teacher_set.update_set_type(set_type)
+        end
+      end
+      assert_equal(TEACHERSET_SETTYPE_ERROR[:code], resp.code)
+      assert_equal(TEACHERSET_SETTYPE_ERROR[:msg], resp.message)
+    end
+  end
+
+
+  describe 'get set-type value' do
+    it 'test book club set set_type value' do
+      resp = @teacher_set.get_set_type('single')
+      assert_equal(TeacherSet::BOOK_CLUB_SET, resp)
+    end
+
+    it 'test topic set set_type value' do
+      resp = @teacher_set.get_set_type('multi')
+      assert_equal(TeacherSet::TOPIC_SET, resp)
+    end
+
+    it 'get set_type value from bib_response' do
+      bib_id = 7899158
+      resp = OpenStruct.new(data: SIERRA_USER["data"], code: 200)
+      @mintest_mock1.expect(:call, resp, [bib_id])
+      resp = nil
+      @teacher_set3.stub :send_request_to_bibs_microservice, @mintest_mock1 do
+        resp = @teacher_set3.get_set_type_value_from_bib_response(bib_id)
+      end
+      assert_equal(TeacherSet::TOPIC_SET, resp)
+    end
+  end
+
+
+  describe 'clear primary_subject column in teacher_set table' do
+    it 'test clean_primary_subject method' do
+      resp = @teacher_set4.clean_primary_subject
+      assert_equal(144, resp)
```

Comment:
What is `144` here?

Reply:
This test doesn't seem to be testing anything about `clean_primary_subject` — length, whitespace, trailing-period, empty string. Probably better to either test each of those, or else just remove this test.